homeassistant/helpers/entity.py

```diff
@@ -642,23 +645,57 @@ def _async_write_ha_state(self) -> None:
         if DATA_CUSTOMIZE in self.hass.data:
             attr.update(self.hass.data[DATA_CUSTOMIZE].get(self.entity_id))
 
-        # Convert temperature if we detect one
-        try:
+        def _convert_temperature(state: str, attr: dict) -> str:
```

Comment:
Why extract to a method ?

Reply:
So we can return the non-modified state from guard clauses instead of modifying the state in a nested if.